scripts/rollover/cross-currency-rollover-new-lender.ts

```diff
@@ -0,0 +1,369 @@
+
+    console.log("Initialize loan...");
+    console.log();
+
+    await originationController
+        .connect(borrower)
+        .initializeLoan(originalLoanTerms, borrowerStruct, originalLender.address, originalSig, sigProperties, []);
+
+    // borrower DAI balance after loan
+    const borrowerDAIBalanceAfter = await dai.balanceOf(borrower.address);
+    // lender DAI balance after loan
+    const lenderDAIBalanceAfter = await dai.balanceOf(originalLender.address);
+    // lender DAI balance after loan is now less
+    const expectedLenderBalance = lenderDAIBalanceBefore.sub(borrowerDAIBalanceAfter);
+    // check if the lender's actual balance matches the expected balance
+    const isLenderBalanceCorrect = lenderDAIBalanceAfter.eq(expectedLenderBalance);
+
+    console.log("Lender DAI balance before loan: ", ethers.utils.formatUnits(lenderDAIBalanceBefore, 18));
+    console.log("Borrower DAI balance before loan: ", ethers.utils.formatUnits(borrowerDAIBalanceBefore, 18));
+    console.log(`Lender DAI balance after loan: ${ethers.utils.formatUnits(expectedLenderBalance, 18)} DAI`);
+    console.log("Borrower DAI balance after loan: ", ethers.utils.formatUnits(borrowerDAIBalanceAfter, 18));
+    console.log("Lender DAI balance is reduced correctly: ", isLenderBalanceCorrect);
+
+    console.log();
+    console.log(
+        `User ${borrower.address} borrowed ${ethers.utils.formatUnits(PRINCIPAL, 18)} DAI at 15% interest from ${
+            originalLender.address
+        } against Nakamigos ${COLLATERAL_ID}`,
+    );
+
+    console.log(SECTION_SEPARATOR);
+
+    // advance time by two weeks into loan
+    await advanceTime(1209600); // 60 * 60 * 24 * 14
+
+    ////////////////////// MIGRATE LOAN TO NEW CURRENCY //////////////////////////////////////////////
+    console.log("Rollover loan from DAI to wETH ...");
+
+    // Calculate amount owed to original lender in the new currency:
+    // (PRINCIPAL + Interest rate ) / swap rate.
+    const loanData: LoanData = await loanCore.getLoan(1);
+    const blockchainTime = new BlockchainTime();
+    const interestAmount = await crossCurrencyRollover.getProratedInterestAmount(
+        loanData.balance,
+        loanData.terms.interestRate,
+        loanData.terms.durationSecs,
+        loanData.startDate,
+        loanData.lastAccrualTimestamp,
+        await blockchainTime.secondsFromNow(3),
+    );
+
+    const amountOwed = PRINCIPAL.add(interestAmount);
+    console.log(
+        "Total amount owed on original loan = Principal + Interest: ",
+        ethers.utils.formatUnits(amountOwed, 18),
+    );
+
+    // price of Dai in wETH at block number: 18852467
+    const price = BigNumber.from("0x018974567f22d0");
+    console.log("Price of DAI in wETH at block number: 18852467: ", ethers.utils.formatUnits(price, 18));
+    // changing the value of NEW_PRINCIPAL will affect whether the borrower will
+    // need to provide additional funds to pay the difference
```

Comment:
Yes this is the case if the new principal value is less than the old pricipal value. It would be nice to be able to change this so that all three scenarions are covered.
- old principal > new pricipal
- old principal = new pricipal
- old principal < new pricipal

My suggestion would be to hardcode what ever the equivlent value of WETH is for 3000 DAI, then you know if you make it higher or lower it will cover all three cases. The `.div(2)` seems very arbitrary.